components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -575,6 +575,20 @@ public Organization addRootOrganization(int tenantId, Organization organization)
         return organization;
     }
 
+    @Override
+    public List<String> getOrganizationsMetaAttributes(Integer limit, String after, String before, String sortOrder,
+                                                       String filter, boolean recursive)
+            throws OrganizationManagementException {
+
+        List<ExpressionNode> expressionNodes = getExpressionNodes(filter, after, before);
+        List<ExpressionNode> filteringByParentIdExpressionNodes = getParentIdExpressionNodes(expressionNodes);
```

Comment:
why do we need parentId based expression node building in org meta attribute filtering service?
AFAIU, this method is used to do some transformation to expression nodes if the filter is based on parent org

Check the same in DAO service as well

Reply:
parentId based expression node building is now removed in all places.